Task_4

georgyaliev opened the pull request:
- Rewrite book structure from hashmap to OpenStruct

- Create OpenStruct structure with headers

- Create new function

- Rewrite parsing function from File.Read to CSV.parse

georgyaliev committed:
Task_4

- Rewrite book structure from hashmap to OpenStruct

- Create OpenStruct structure with headers

- Create new function

- Rewrite parsing function from File.Read to CSV.parse

georgyaliev committed:
Return and rewrite function for creationg book with OpenStruct object

georgyaliev committed:
All lost functions were return and modernized for OpenStruct class

georgyaliev committed:
Month_statistics function was separated (output function was excluded) and now output use function that returning values

georgyaliev committed:
Identation was fixed

georgyaliev commented:
> Naming is not the best, try something different

Global or in some function?

georgyaliev commented:
> Also fix the indentations

fixed

georgyaliev commented:
> Where all functions? Why you delete them?

All function were returned and modernized for OpenStruct

goblynwtf commented on books.rb:
```diff
@@ -1,56 +1,63 @@
-    else
-        Date.parse("01-01"+date)
-    end
+def make_date(date)
```
Tabs? Again? Cmon...

goblynwtf commented on books.rb:
```diff
@@ -1,56 +1,63 @@
+	if (date.size>4) 
+		Date.parse(date)
+	else (date.size<5)
+  	Date.parse("01-01"+date) 
```
Indentation and whitespaces all-around again...

goblynwtf commented on books.rb:
```diff
@@ -1,56 +1,63 @@
+books = CSV.parse(File.read("books.txt"), row_sep: "\n", col_sep: "|")
+           .map { |book| OpenStruct.new(make_openstruct(book))}
 
 def cool_print(array_of_books)
```
Wrong, this function should know how to print book, why there is iteration over array?

goblynwtf commented on books.rb:
```diff
@@ -1,56 +1,63 @@
+	book_array.select {|book| book.date.year==year}
 end
 
 def novels(book_array)
```
Unificate this function. Told you many times.

goblynwtf commented on books.rb:
```diff
@@ -1,56 +1,63 @@
-    book_array.select {|book| book[:genres].include?("Novels")}
-              .sort_by{ |book| book[:date] }
+	book_array.select {|book| book.genres.include?("Novels")}
+	.sort_by{ |book| book.date }
```
And why there is sorting? Name of this function not good, but at least you can understand that there is novels, but why there is also sorting? What if I don't need sorting?

goblynwtf commented on books.rb:
```diff
@@ -1,56 +1,63 @@
-    book_array.map{ |book| book[:author].split(' ') }
-              .sort_by{ |name| name.last }
-              .uniq
+	book_array.map{ |book| book.author.split(' ') }
```
Again, function does too much.

georgyaliev committed:
I changen name of novels-function

georgyaliev committed:
Identationt trying

georgyaliev committed:
Fixed identatition

georgyaliev committed:
reidentation

georgyaliev committed:
- Rewrite cool print function (now it uses openstruct of book instead of array)
- Fixed indentation
- Rewrite ouputs
- Rewrite functions of selectiong objects (Now functions do only 1 thing)

georgyaliev committed:
Rewrite all search function to one with different parameters
